File: nova-api/src/main/java/xyz/xenondevs/nova/api/block/NovaTileEntityState.java
package xyz.xenondevs.nova.api.block;

import org.jetbrains.annotations.NotNull;
import xyz.xenondevs.nova.api.tileentity.TileEntity;

public interface NovaTileEntityState extends NovaBlockState {
    
    /**
     * Gets the {@link TileEntity} represented by this {@link NovaTileEntityState}.
     *
     * @return The {@link TileEntity} represented by this {@link NovaTileEntityState}.
     */
    @NotNull TileEntity getTileEntity();
    
}

File: nova-api/src/main/kotlin/xyz/xenondevs/nova/api/block/NovaTileEntityState.kt
package xyz.xenondevs.nova.api.block

import xyz.xenondevs.nova.api.tileentity.TileEntity

interface NovaTileEntityState : NovaBlockState {
    
    /**
     * The tile-entity represented by this [NovaTileEntityState].
     */
    val tileEntity: TileEntity
    
}